pkg/manager/manager.go

```diff
@@ -229,6 +230,11 @@ type Options struct {
 
 	// Functions to all for a user to customize the values that will be injected.
 
+	// WebhookTLSConfig is a *tls.Config object that will be used in place of
+	// the one normally created by reading certs in the CertDir.  If both
+	// CertDir and this are specified CertDir will be ignored
```

Comment:
Same comment as above about making it an error to specify both CertDir and WebhookTLSConfig.

Perhaps we should also:
1. Introduce an exported function (maybe in the `certwatcher` package?) that can build the same `tls.Config` that is built when `CertDir` and any other user-accessible TLS config settings are used.
2. Deprecate `CertDir` and any related fields, pointing folks to to the function exported from (1)
3. In a later release, remove `CertDir` and related fields.

Reply:
Hmm. Thinking about this more, it might be too much to ask users to setup the cert watcher and configure it as a runnable in the manager to get equivalent behavior to what they have now.

Thoughts @vincepri @alvaroaleman @varshaprasad96?

Reply:
> Hmm. Thinking about this more, it might be too much to ask users to setup the cert watcher and configure it as a runnable in the manager to get equivalent behavior to what they have now.

Yeah. How about instead using functional like this `opts ...func(*tls.Config)` and if we ultimatively have file(s) in there, start the certwatcher?

Reply:
> How about instead using functional opts like this `opts ...func(*tls.Config)`

Sorry, I'm not sure I fully understand that pattern - can you point me at an example/documentation?

Reply:
It's this pattern: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

But I'm not sure exactly what @alvaroaleman has in mind for where to put this.